File: Application/Backend/CodeNB.Repository/Concrete/AuthenticationRepository.cs
using CodeNB.Model;
using CodeNB.Types;
using DAL;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CodeNB.Repository
{
    public class AuthenticationRepository : IAuthenticationRepository
    {
        private readonly IDataAccess _db;
        public AuthenticationRepository(IDataAccess dataAccess) => _db = dataAccess;

        public async Task<byte[]?> GetUserSalt(int? id)
        {
            var salt = await _db.ExecuteScalarAsync("spGetSaltByEmployeeId", [new("@ID", SqlDbType.Int, id)]);

            if (salt == null)
                return null;

            return (byte[])salt;
        }

        public async Task<LoginResultDto?> Login(LoginDto user)
        {
            DataTable dt = await _db.ExecuteAsync("spLogin", [
                new Parm("@ID", SqlDbType.Int, Convert.ToInt32(user.Username)),
                new Parm("@HashedPassword", SqlDbType.NVarChar, user.Password, 64)
            ]);

            if (dt.Rows.Count == 0)
                return null;

            DataRow row = dt.Rows[0];

            return new LoginResultDto
            {
                ID = (int)row["ID"],
                FirstName = row["FirstName"].ToString(),
                LastName = row["LastName"].ToString(),
                Role = row["Role"].ToString(),
            };
        }
    }
}

File: Application/CodeNB/app/src/main/java/com/example/codenb/repository/AuthenticationRepository.kt
package com.example.codenb.repository

import com.example.codenb.data.AuthService
import com.example.codenb.model.LoginDto
import com.example.codenb.model.LoginResultDto
import retrofit2.Response

interface AuthRepository {
    suspend fun login(loginRequest: LoginDto): Response<LoginResultDto>
}

class RemoteAuthRepository(
    private val authService: AuthService
) : AuthRepository {
    override suspend fun login(loginRequest: LoginDto): Response<LoginResultDto> {
        return authService.login(loginRequest)
    }
}